File: app/src/main/java/com/example/data/model/Donation.kt
package com.example.data.model

import java.io.Serializable

data class Donation(
    val id: Int = 0,
    val donorId: Int,
    val donorName: String,
    val donorPhone: String,
    val donorProfileImage: String? = null,
    val requestId: Int,
    val status: String = "accepted",
    val timestamp: Long = System.currentTimeMillis()
) : Serializable

File: blood-connect-server/models/Donation.js
const { DataTypes } = require('sequelize');
const { sequelize }  = require('../config/database');

const Donation = sequelize.define('Donation', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true
  },
  donor_id: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  donor_name: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  donor_phone: {
    type: DataTypes.STRING(20),
    allowNull: false
  },
  donor_profile_image: {
    type: DataTypes.TEXT,   // Cloudinary URL
    allowNull: true
  },
  request_id: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  status: {
    type: DataTypes.ENUM('accepted', 'rejected', 'completed'),
    defaultValue: 'accepted'
  }
}, {
  tableName: 'donations',
  timestamps: true,
  createdAt: 'timestamp',
  updatedAt: false
});

module.exports = Donation;
